### DISCUSSION TITLE
Week 6 Accessibility

### DISCUSSION THREAD

**Discussion by jakegros** [USU-ITLS-x265-23-summer/class-repo#129] [General]
<h1>Accessibility Reflections</h1>
<h2>Thoughts and impressions of vid/reading:</h2>
The video was great juxtaposition, especially when the kid was pointing and the mother was trying to stop him from judging.  It adds perspective and helps to frame our thinking from another's experience.  I really enjoyed the video and actually chuckled a few times realizing how absurd some things have been designed are the "norm."

Thinking about how we build our world to conform to some but not others is interesting and becomes cumbersome when trying to add a "fix" as an afterthought. There are little shifts we can make that allow access for those that need it but also helps everyone.  For example, closed captioning.  It is necessary for some but helpful for all.  Can you thing of a time you used CC?  I do on almost everything (mainly because I will never know what is going on while watching something when my three year old is around without cc 🤣).
Most importantly, even though the developed world has severely limited who can participate, that doesn't mean there is nothing we can do about it.  It starts with being mindful of others' experiences (UX) and designing with accessibility in mind.  I cringe when I think about content I claimed was "accessible." 

<h2>Instructional approaches reflection:</h2>
I feel like the CSS Color Markers activity add better explanations and background for each step.  Perhaps I am just getting a better grasp on the exercises. I appreciate the activities giving me more practice.  The lesson projects are different enough to learn new things and applications, yet keeps laying a foundation that allows me to become more familiar. The repetitive manipulation was helpful.  I like how it compared/contrasted some dos and don'ts, like what colors are too bright, can be distracting, and explaining better practices for design. 

The Accessibility Quiz took longer.  It was a bit more complicated than expected since the lessons assumed I remembered everything previously taught 🙀.  Some of the steps included several things that were walked through in previous projects and not gone over for this project.
Ideally, I would like to design things that allow for multiple inputs or ways to interact with what I develop.

<h2>Project updates:</h2>
-Website / Game

Some technical challenges I will run into and how to overcome: 
Which game to create?  Where to even start with a website? I suppose I should start small and break it down into a manageable task. I will continue to explore the GitHub pages resources. It has been a challenge just to debug during our lessons while learning.  Thus, what to do when my code is not working as expected, not knowing where error is.  I can reach out to others to get their take on it or try debugging with the strategies that folks in this course have mentioned this week.
In addition to exploring resources with tutorials and lessons, I would start looking at code for simple games/websites and seeing which areas of it I already understand from our lessons and which areas I need to figure out.  I would then look at manipulating it in simple ways, then look at more complex ones.  The best way is to start is to dive in, make mistakes, develop community, and work in consistent, small chunks.

**Comment by metasean** [comment]
> Thinking about how we build our world to conform to some but not others is interesting and becomes cumbersome when trying to add a "fix" as an afterthought.

Very well said!

> Can you thing of a time you used CC? I do on almost everything 
Ditto.  I also have a buddy who is a linguist, and will switch the CC to another language to help him stay proficient, learn slang, etc.  I'm pretty sure it's a recommendation for ESL learners do the same thing.

> I cringe when I think about content I claimed was "accessible." 

Sadly, it's hard to know all the things that one should do, not just in web development, but in other areas also.  To paraphrase: We do what we can, to the best of our ability, and when we learn to do better, we can do better.

> The Accessibility Quiz took longer. It was a bit more complicated than expected since the lessons assumed I remembered everything previously taught 🙀. Some of the steps included several things that were walked through in previous projects and not gone over for this project.

This was a hard choice for me, but there were two reasons for it.  First, I absolutely wanted y'all to cover accessibility.  Second, as you're working on projects, you're going to have to go and learn how to do things on your own, this was a relatively safe assignment to do that on, since you'd have the fallback of the FFC community.

>The best way is to start is to dive in, make mistakes, develop community, and work in consistent, small chunks.

100%!
